api/dataset.go

```diff
@@ -51,7 +73,7 @@ func (api *DatasetAPI) getDatasets(w http.ResponseWriter, r *http.Request) {
 
 			b, err = json.Marshal(datasets)
 			if err != nil {
-				logError(ctx, errors.WithMessage(err, "api endpoint getDatasets failed to marshal dataset resource into bytes"), logData)
+				audit.LogError(ctx, errors.WithMessage(err, "api endpoint getDatasets failed to marshal dataset resource into bytes"), logData)
```

Comment:
Instead of repeating this `if err != nil` block twice in the if and else block, just do it once after the else clause. I.e

```go
if authorised {
    //
} else {
   //
}
if err != nil {
    audit.LogError(ctx, errors.WithMessage(err, "api endpoint getDatasets failed to marshal dataset resource into bytes"), logData)
}
return b, err
```